### ISSUE TITLE
Tower and blade mode shape coefficients in ElastoDyn

### ISSUE THREAD

**Issue by wiky231** [IEAWindTask37/IEA-22-280-RWT#87]
How are the tower and blade mode shape polynomial coefficients used in the motion equations in ElastoDyn?

**Comment by axlecky** [comment]
and also, is it possible to get the BModes input files for the 22MW RWT?

**Comment by gbarter** [comment]
Hello,

Your first question is best addressed in the ElastoDyn user guide and theory manual [here](https://openfast.readthedocs.io/en/dev/source/user/elastodyn/index.html).  On the second question, BModes is deprecated software and was not used in the creation of the reference turbine design.  I understand it is still preferred by a good number of OpenFAST users. For this effort, WISDEM, via pyFrame3DD, outputs the mode shapes that are processed for the polynomial coefficients instead.  You could probably create the BModes input file yourself from the tower information in the [tabular excel file](https://github.com/IEAWindTask37/IEA-22-280-RWT/tree/main/Documentation).

**Comment by axlecky** [comment]
Hi Garrett,

Thanks for the swift response. However, I have tried looking into the ElastoDyn documentation (FASTKinematics, FASTKinetics, FASTMotions, FASTLoads) and could not find an exact link to the mode shape coefficients. 

I managed to find equations in FASTKinematics that relate to the tower FA and SS modes, such as the position vectors (shown in the following screenshot): 

![FASTKinematics equation](https://github.com/IEAWindTask37/IEA-22-280-RWT/assets/103893027/9c65b56e-d644-426a-8a8e-a26d64cc94bb)

What does the following term φ_1^TFA (h) q_TFA1 represent in the equation? 

Thanks again and I look forward to your response.

**Comment by jjonkman** [comment]
Hi @wiky231,

In this equation, `phi^TFA_1(h)` is the 1st tower fore-aft bending mode shape (derived from the user-specified polynomial coefficients) and `q_TFA1` is the displacement of the 1st tower fore-aft bending degree of freedom.

FYI: Rather than here, I'd recommend asking ElastoDyn/OpenFAST-related questions either on our forum (https://forums.nrel.gov) or in OpenFAST discussions (https://github.com/OpenFAST/openfast/discussions).

Best regards,

**Comment by wiky231** [comment]
Thanks Jonkman,

I am currently re-calculating the 22MW Tower (Fixed) mode shapes using BModes, however, if I compare it with the monopile tower ElastoDyn_Tower file posted on GitHub, there is some similarity between the first F-A mode shapes, but the second F-A mode shape differ alot. (Please see the image below for the BModes coefficients for the second F-A mode shape)

![image](https://github.com/IEAWindTask37/IEA-22-280-RWT/assets/103893027/256d447b-ecf3-4412-bb0c-22bb0d2670c1)

For the BModes input, I input the inertia values from the 22MW tabular excel file and used parallel axis theorem to compute the inertias about the tower top axes. (Please see below image). The other values in the section file such as stiffness and inertias are set as per the tabular excel file and the torsional and axial stiffnesses are also set to very high values as recommended in other posts.

![image](https://github.com/IEAWindTask37/IEA-22-280-RWT/assets/103893027/4a3cd39a-278a-4557-8e69-2339563fe39e)

Can I know if there is any reason for this variation between the 2nd mode shape coefficients for BModes vs WISDEM?

**Comment by jjonkman** [comment]
Dear @wiky231,

Certainly the tower-base boundary condition (flexibility of the monopile) can impact the tower mode shapes, especially the higher mode shapes.  But I was not the one to derive the tower mode shapes for the IEA Wind 22-MW reference wind turbine, so, I can't comment on how the approach used differs from yours.  I'll let developers of the IEA Wind 22-MW reference wind turbine respond.

Best regards,

**Comment by gbarter** [comment]
BModes and Frame3DD do not take the exact same modeling approach. The values you derived for mass, moments of inertia, and center of mass for the rotor-nacelle assembly and the distributed tower properties are likely close, but not exact, to the values used in the design optimization.  I would not expect the polynomial coefficients to match exactly.

**Comment by gbarter** [comment]
I am re-opening this to make the comment that you were correct that the second mode shapes were not quite right.  My apologies for trying to explain away the issues you raised.  As I dug into the code further, I saw a possibility where mistakes could arise.  After improving the logic in our WISDEM code, I regenerated the mode shapes and saw differences in the second mode coefficients especially.  This has been corrected in PRs for this repository (#93)and the 15-MW repository (https://github.com/IEAWindTask37/IEA-15-240-RWT/pull/214).
